File: dotnet/SlimLO/SlimLOException.cs
namespace SlimLO;

/// <summary>
/// Exception thrown by SlimLO operations when a conversion or initialization fails.
/// </summary>
public class SlimLOException : Exception
{
    /// <summary>
    /// The native error code associated with the failure.
    /// </summary>
    public SlimLOErrorCode ErrorCode { get; }

    public SlimLOException(string message, SlimLOErrorCode errorCode = SlimLOErrorCode.Unknown)
        : base(message)
    {
        ErrorCode = errorCode;
    }

    public SlimLOException(string message, SlimLOErrorCode errorCode, Exception innerException)
        : base(message, innerException)
    {
        ErrorCode = errorCode;
    }
}

File: java/slimlo/src/main/java/com/slimlo/SlimLOException.java
package com.slimlo;

/**
 * Exception thrown by SlimLO operations.
 */
public class SlimLOException extends RuntimeException {

    private final SlimLOErrorCode errorCode;

    public SlimLOException(String message, SlimLOErrorCode errorCode) {
        super(message);
        this.errorCode = errorCode;
    }

    public SlimLOException(String message, SlimLOErrorCode errorCode, Throwable cause) {
        super(message, cause);
        this.errorCode = errorCode;
    }

    public SlimLOErrorCode getErrorCode() {
        return errorCode;
    }
}
